include/h2o.h

```diff
@@ -618,8 +618,18 @@ typedef struct st_h2o_generator_t {
     void (*stop)(struct st_h2o_generator_t *self, h2o_req_t *req);
 } h2o_generator_t;
 
-typedef int (*h2o_ostream_pull_cb)(h2o_generator_t *generator, h2o_req_t *req, h2o_iovec_t *buf);
+enum h2o_stream_send_state {
+    H2O_STREAM_SEND_STATE_IN_PROGRESS,
+    H2O_STREAM_SEND_STATE_FINAL,
+    H2O_STREAM_SEND_STATE_ERROR,
+};
```

Comment:
Please use typedef as well as adding `_t` prefix, following the patterns found in other parts of the code (e.g. https://github.com/h2o/h2o/blob/master/include/h2o/http2_internal.h#L138).

Also, how about omitting `_stream` from the typename and the states, considering the fact that the function that sends this code is named `h2o_send`? Note that the concept of _stream_ is specific to HTTP/2.